### PR TITLE
Added ProjectInfo.

### PR DESCRIPTION
**Pull request by mderka**

### CODE REVIEW

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+import static com.google.api.client.repackaged.com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * The class that encapsulates information about a project in Google Cloud DNS. A project is a top
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: I would say something like:

This class provides the Google Cloud DNS information associated with this project.
....

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+  private static final long serialVersionUID = 8696578863323485036L;
+  private final String id;
+  private final BigInteger number;
+  private final Quota quota;
+
+  /**
+   * This class represents quotas assigned to the {@code ProjectInfo}.
+   *
+   * @see <a href="https://cloud.google.com/dns/api/v1/projects">Google Cloud DNS documentation</a>
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: https://cloud.google.com/dns/api/v1/projects#quota ?

Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: Would be good if we can provide pointers on how to change the quota for DNS related settings.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+   */
+  public static class Quota {
+
+    private Integer zones;
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: If values are never null lets make them primitive.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    /**
+     * Returns the maximum allowed size for total records in one ChangesRequest in bytes.
+     */
+    public Integer totalRrdataSizePerChange() {
+      return totalRrdataSizePerChange;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      return (other instanceof Quota) && this.toPb().equals(((Quota) other).toPb());
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: per @ajkannan  earlier Nit. "this" is redundant in this context.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+    private Quota quota;
+
+    private Builder() {
+    }
+
+    /**
+     * Sets an id of the project.
+     */
+    Builder id(String id) {
+      this.id = checkNotNull(id);
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: is that the project id?

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+      return this;
+    }
+
+    /**
+     * Sets a number of the project.
+     */
+    Builder number(BigInteger number) {
+      this.number = checkNotNull(number);
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: is that the project number?

If so (and for the above), I find it strange to have this information specific to DNS.

Project id would be available from the options, though maybe we should have part of the common Service api
a way to get the id (and number), which could be null for some services.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java (hunk 1 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: Done.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java (hunk 3 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: Done.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java (hunk 4 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: fixed

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java (hunk 5 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: Yes. Getters removed as per discussion.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java (hunk 6 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: Getters removed as per discussion.

**Comment by mderka** [comment]
I implemented the comments. Ready for another pass.

**Comment by aozarov** [comment]
Looks good. Please change the replaces the `@code` with `@link` in the places we discussed (where there was already another `@link` entry) and feel free to merge after.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+    /**
+     * Returns the maximum allowed number of {@link DnsRecord}s to add per {@link ChangesRequest}.
+     */
+    public int rrsetAdditionsPerChange() {
+      return rrsetAdditionsPerChange;
+    }
+
+    /**
+     * Returns the maximum allowed number of {@link DnsRecord}s to delete per {@link
+     * ChangesRequest}.
```
Comment (ajkannan) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: ChangesRequest --> ChangeRequest

Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java]: Yes, fixed.

**Pull request merged**